### DISCUSSION TITLE
Feature Suggestion/Request: Sync Timestamp

### DISCUSSION THREAD

**Discussion by eswiig** [vzhd1701/gridplayer#47] [Ideas]
Not sure if this should be filed under Issues or Discussions, please feel free to delete if in the wrong section.

Suggestion
Due to the delay in opening several videos, there will often be 1+ second difference in the playback. Yes, it's possible to skip +15/-15 secs ahead/back, but I think what the player needs is a "Sync all video playback".

Let's say you got 3 Video Windows

Video 1: 00:00:01
Video 2: 00:00:02
Video 3: 00:00:03

Suggestion either a right click video window, or a button that lets you enter the timestamp to sync to by input: XX:XX:XX or maybe even XX:XX is enough

I would suggest an option to right click Video1 with an option to "Sync players", meaning all videos would sync to timestamp 00:00:01.

If you right clicked Video 2 and chose "Sync Players" all videos would sync timestamp to 00:00:02
Video 3, 00:00:03.

Another Example: Fast forward a video to say 04:00:00 and sync, and all other players would fast forward to 04:00:00 as well.

Possible complication: if a videos duration is less than the sync time stamp though.

Not sure if this is possible, but would appreciate if it could be considered as a feature request.

**Comment by hiroo916** [comment]
I second this suggestion.

I primarily want to use this to compare multiple encodes of the same video to look at differences in encoding quality.  To do this, obviously the videos need to be in sync.

The feature as suggested would help a lot.

An additional feature suggestion would be a "Keep all videos in sync" so that one click in the time line of any video would also jump the other videos to sync to that timestamp.

**Reply by brunetton** [reply]
Exately !

**Comment by dsmmr** [comment]
+1!
A feature I would like to see this feature in this (already awesome) software. Setting Sync markers to a user-defined time would be really helpful. 
I regularly have Videos with arbitrary time before an actual event that happens in all videos. Then sync by % or time does not work.

**Comment by Prog-Newb** [comment]
This is a great platform to build. These features and most anything can likely be added by the individual. I stripped / minimized prompts,  menus,  added dark theme and delete functions. Once inside command lines are not hard to follow. ai can help with the coding, you have to be os savy to fill in the blanks, dependencies, shortcuts, and smooth it over. I ended with something like the portable version but modded. Opens and runs as smooth as the installer exe. 

Also If you go to advanced and set the hardware decoder to 1 or 2 per process, it handles multiple vids much better. I know they say 4 is optimum, but I have not seen any benefit to increased value except freeze buffer wait freeze buffer wait. 2 below that all disappeared for me in my testing. My system is uptodate mid-upper power class.
